.github/workflows/ci-cd.yml

```diff
@@ -127,12 +124,36 @@ jobs:
         with:
```

Comment:
Does this step both build dists and creates a GitHub Release? Can it be uncoupled?

I strongly suggest building the dists before the `check_if_should_release` job, unconditionally, so they're smoke-tested on each push/PR. You can use `python -Im build` as shown here https://packaging.python.org/en/latest/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/#checking-out-the-project-and-building-distributions. Then, it's a good idea to run `python -Im twine check --strict dists/*` after building.

This would make sure that any problems with the dists are caught way before somebody decides to tag a release. As early as possible. In the long run, it'd enable testing the project as it's actually installed by the end-users, as opposed to testing a Git checkout which may differ from what the users would get in their `site-packages/`.

Reply:
> I strongly suggest building the dists before the `check_if_should_release` job, unconditionally, so they're smoke-tested on each push/PR. You can use `python -Im build` as shown here https://packaging.python.org/en/latest/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/#checking-out-the-project-and-building-distributions. Then, it's a good idea to run `python -Im twine check --strict dists/*` after building.
> 
> This would make sure that any problems with the dists are caught way before somebody decides to tag a release. As early as possible. In the long run, it'd enable testing the project as it's actually installed by the end-users, as opposed to testing a Git checkout which may differ from what the users would get in their `site-packages/`.

The test_release should build it every time if it’s not actual release

Reply:
> Does this step both build dists and creates a GitHub Release? Can it be uncoupled?

I think at some point it used to do that, but they changed it in version eight to only build the dist and the comment is stale.  Will validate that

Reply:
As of version 8.x they split it, and the action doesn't create the release and upload to github until `python-semantic-release/upload-to-gh-release`

Reply:
It looks like `test_release` doesn't actually try to build the dist if it detects no release is needed.

https://github.com/aio-libs/aiohappyeyeballs/actions/runs/7442151974/job/20245242989?pr=44

There doesn't appear to be an option to build to force build the distribution with `python-semantic-release`

Reply:
I've add a similar poetry based check in 
https://github.com/aio-libs/aiohappyeyeballs/actions/runs/7442259204/job/20245480361?pr=44